db/collab.go

```diff
@@ -0,0 +1,58 @@
+package db
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"github.com/uclaacm/teach-la-go-backend/httpext"
+	"golang.org/x/net/websocket"
+)
+
+// Session describes a collaborative coding environment.
+type Session struct {
+	// Map UIDs to their websocket.Conn
+	Conns map[string]*websocket.Conn `json:"conns"`
+}
+
+// Maps session IDs to Session object
+var sessions map[string]Session
+
+func init() {
+	sessions = make(map[string]Session)
+}
+
+func (d *DB) CreateCollab(c echo.Context) error {
+	var body struct {
+		UID string `json:"uid"`
+	}
+	if err := httpext.RequestBodyTo(c.Request(), &body); err != nil {
+		return c.String(http.StatusInternalServerError, "failed to read request body")
+	}
+
+	sessionId := uuid.New().String()
+	sessions[sessionId] = Session{
+		Conns: make(map[string]*websocket.Conn),
+	}
+
+	// Kill session if no connections every minute
+	go func() {
+		ticker := time.NewTicker(time.Minute)
+		for {
```

Comment:
Just elaborating on the error returned by the linter, this can be made simpler with `for range ticker.C { ... }`.

For notes on future design: later on, more complicated behavior like checking for connections will be moved to the actual websocket endpoint for a given session. Though goroutines are cheap, it would be wise to avoid spawning more than absolutely necessary. We'll also need to lock the sessions variable since the `delete` function is not thread-safe.